internal/pkg/grpc/tls/tls.go

```diff
@@ -24,26 +24,70 @@ const (
 	ServerOnly
 )
 
-// NewConfig initializes a TLS config for the specified type.
-func NewConfig(t Type, data *userdata.OSSecurity) (config *tls.Config, err error) {
-	certPool := x509.NewCertPool()
-	if ok := certPool.AppendCertsFromPEM(data.CA.Crt); !ok {
-		return nil, fmt.Errorf("failed to append client certs")
+// ConfigOptionFunc describes a configuration option function for the TLS config
+type ConfigOptionFunc func(*tls.Config) error
```

Comment:
👍

Reply:
This is actually a little silly, right now, since it returns a standard public object which can then be directly modified.  However, I suspect that may change in the future, or that we may want to add more project-specific options.